Fix vari

LoreRizzetto opened the pull request:
- aggiunto include mancante
- aggiunto JSON.generate ove necessario
- rimossa interpolazione superflua

cristianogregnanin commented on test.rb:
```diff
@@ -51,7 +52,7 @@
     end
 
-    describe "GET /products/#{$id}" do
+    describe "GET /products/{id}" do
```
$

LoreRizzetto replied:
- rimossa interpolazione superflua

quando viene interpolata la stringa $id non è definita (quindi nil) quindi non ha effetto:
```ruby
irb(main):001:0> "id è #{$id}";
=> "id è "
```
se vuole vederlo nel codice basta aggiungere`it { expect(3).to eq(2) }` dentro un blocco che è definito usando l'id e noterà che il nome del blocco non include l'id:
```
Failures:

  1) Test GET /products/ is expected to eq 2
     Failure/Error: it { expect(3).to eq(2) }

       expected: 2
            got: 3

       (compared using ==)
```

senza interpolazione secondo me è più chiaro:
```
Failures:

  1) Test GET /products/{id} is expected to eq 2
     Failure/Error: it { expect(3).to eq(2) }

       expected: 2
            got: 3

       (compared using ==)
```

cristianogregnanin replied:
ok però allora usiamo :id al posto di {id}

cristianogregnanin commented on test.rb:
```diff
@@ -12,7 +13,7 @@
     describe "POST /products" do
       before (:each) do
-        @response = Faraday.post("http://#{address}:#{port}/products", $body)
+        @response = Faraday.post("http://#{address}:#{port}/products", JSON.generate($body))
```
Content-Type è automatico o bisogna esplicitarlo? 
Faraday uses convention to assume that any content you're sending in a POST request is of the type application/x-www-form-urlencoded

A me funziona senza JSON.generate, però il mio backend è fatto in rails.. magari fa qualche magia

LoreRizzetto replied:
probabilmente dev'essere esplicito ma non credo importi visto che jsonapi definisce tutta la comunicazione come json (quindi sia client che server possono assumere che sia sempre json anche se non specificato).

Anche il client non controlla se il content type è json:
https://github.com/cristianogregnanin/json_api_test/blob/de65534b345f474effbb858efc66b5b6176772a0/test.rb#L15-L16

https://github.com/cristianogregnanin/json_api_test/blob/de65534b345f474effbb858efc66b5b6176772a0/test.rb#L38-L39

https://github.com/cristianogregnanin/json_api_test/blob/de65534b345f474effbb858efc66b5b6176772a0/test.rb#L56-L57

cristianogregnanin replied:
ok, quindi JSON.parse lo mettiamo o no? a me funziona anche senza. Per me meno cose si mettono meglio è

LoreRizzetto replied:
non usare JSON.generate non invia json, invia un form urlencoded (la versione che c'è ora)

usare JSON.generate senza cambiare gli header invia json con un content type errato (la pr che ho fatto)

usare JSON.generate e indicare gli header esplicitamente invia json con il content type corretto (questo: `@response = Faraday.post("http://#{address}:#{port}/products", JSON.generate($body), {'Content-Type'=>'application/json'})`)

questa è la versione "corretta" ma non specificare il content type (quindi quello che ho inviato nella pr) non dovrebbe essere un problema

cristianogregnanin replied:
@response = Faraday.post("http://#{address}:#{port}/products", JSON.generate($body), {'Content-Type'=>'application/json'}))    usiamo questa

cristianogregnanin commented on test.rb:
```diff
@@ -1,8 +1,9 @@
 require "json"
 require "faraday"
+require "rspec"
```
A me funziona anche senza il require

LoreRizzetto replied:
se lo invoca con rspec sì, se lo esegue con ruby fallisce perché RSpec non è definito. Potrebbe essere una buona idea controllare se è stato invocato con ruby e, in caso, mostrare un messaggio di errore

cristianogregnanin replied:
beh ma io direi di invocarlo con rspec non con ruby

cristianogregnanin reviewed (commented):
ok, però manca :id

Commit:
Fix vari

- aggiunto include mancante
- aggiunto JSON.generate ove necessario
- rimossa interpolazione superflua

- aggiunto mike ehrmantraut:
⠀⠀⠀⠀⠀⠀⠀⠀⣀⣴⣶⣿⣿⣿⣿⣿⣿⣿⣶⣦⣀⠀⠀⠀⠀⠀⠀⠀
⠀⠀⠀⠀⠀⠀⣤⣾⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣄⠀⠀⠀⠀⠀
⠀⠀⠀⠀⢀⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣧⠀⠀⠀⢠
⠀⠀⠀⠀⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣟⣛⣻⣿⣿⣟⣿⣿⣿⣷⠀⠀⠀
⠀⠀⠀⠀⣿⣿⣿⣿⣿⣿⣿⣿⣿⣫⣽⣾⣻⣾⣿⣿⣿⣿⡿⣿⣿⠀⠀⠀
⠀⠀⠀⢰⣿⣿⣻⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⠻⡿⠿⠟⠛⣟⣿⣽⠀⠀⠀
⠀⠀⠀⠸⣿⣿⣿⣷⣿⣿⣿⣿⡿⠍⠈⠀⠁⣴⡆⠀⠀⠠⢭⣮⣿⡶⠀⠀
⠀⡴⠲⣦⢽⣿⣿⣿⣿⣿⣟⣩⣨⣀⡄⣐⣾⣿⣿⣇⠠⣷⣶⣿⣿⡠⠁⠀
⠀⠃⢀⡄⠀⢻⣿⣿⣿⣿⣽⢿⣿⣯⣾⣿⣿⣿⣿⣿⢿⣿⣿⡟⣿⠀⠀⠀
⠀⠀⠣⠧⠀⢿⣿⣿⣿⣿⣿⣿⣿⣿⠟⢸⣿⠿⠿⠿⣧⠙⣿⣿⡿⠀⠀⠀
⠀⠀⠀⠁⠼⣒⡿⣿⣿⣿⣿⣿⣿⣿⣠⣬⠀⠀⠀⠀⣾⣷⡈⣿⡇⠀⠀⠀
⠀⠀⠀⠀⠀⠉⢳⣿⣿⣿⣿⣿⣿⣿⢟⠗⠼⠖⠒⠔⠉⠉⠻⣿⠇⠀⠀⠀
⠀⠀⠀⠀⠀⠀⠈⣻⡿⣿⣿⣿⣿⡿⡀⣤⡄⠸⣰⣾⡒⣷⣴⣿⠀⠀⠀⠀
⠀⠀⠀⠀⠀⠀⠂⢸⡗⡄⠘⠭⣭⣷⣿⣮⣠⣌⣫⣿⣷⣿⣿⠃⠀⠈⠀⠀
⠀⠀⠀⠀⠀⠈⠀⢸⣿⣾⣷⣦⡿⣿⣿⣿⡿⢻⠞⣹⣿⣿⠏⠀⠀⠀⠀⠀
⠀⠀⠀⠀⠀⢘⠀⠘⢻⡿⢿⣋⣤⣤⠌⠉⠛⠛⠀⠈⠉⠁⠀⠀⠀⠀⠀⡀

Commit:
Test if being run with 'ruby' or 'rspec'
